packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -376,4 +376,13 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 
 		return aggregatedInsightsByTimeParser.parse(rawRows);
 	}
+
+	async pruneOldData(maxAgeInDays: number): Promise<{ affected: number }> {
```

Comment:
Can it happen that we have so much volume to prune that we need to speed up the next cycle? Executions pruning uses a timeout instead of a timer for this reason.

Reply:
You're right, it's safer to use a setTimeout. I changed it